warehouse/upload.go

```diff
@@ -824,19 +828,76 @@ func (job *UploadJobT) updateTableSchema(tName string, tableSchemaDiff warehouse
 		return nil
 	}
 
-	if err := job.addColumnsToWarehouse(tName, tableSchemaDiff.ColumnMap); err != nil {
-		return err
+	if err = job.addColumnsToWarehouse(tName, tableSchemaDiff.ColumnMap); err != nil {
+		return fmt.Errorf("adding columns to warehouse: %w", err)
+	}
+
+	if err = job.alterColumnsToWarehouse(tName, tableSchemaDiff.AlteredColumnMap); err != nil {
+		return fmt.Errorf("altering columns to warehouse: %w", err)
 	}
 
-	for _, columnName := range tableSchemaDiff.StringColumnsToBeAlteredToText {
-		err = job.whManager.AlterColumn(tName, columnName, "text")
+	return nil
+}
+
+func (job *UploadJobT) alterColumnsToWarehouse(tName string, columnsMap map[string]string) error {
+	var responseToAlerta []model.AlterTableResponse
+	var errs []error
+
+	for columnName, columnType := range columnsMap {
+		res, err := job.whManager.AlterColumn(tName, columnName, columnType)
 		if err != nil {
-			pkgLogger.Errorf("Altering column %s in table: %s.%s failed. Error: %v", columnName, job.warehouse.Namespace, tName, err)
-			break
+			errs = append(errs, err)
+			continue
+		}
+
+		if res.IsDependent {
+			responseToAlerta = append(responseToAlerta, res)
+			continue
 		}
+
+		pkgLogger.Infof(`
+			[WH]: Altered column %s of type %s in table %s in namespace %s of destination %s:%s
+		`,
+			columnName,
+			columnType,
+			tName,
+			job.warehouse.Namespace,
+			job.warehouse.Type,
+			job.warehouse.Destination.ID,
+		)
 	}
 
-	return err
+	if len(responseToAlerta) > 0 {
+		queries := make([]string, len(responseToAlerta))
+		for i, res := range responseToAlerta {
+			queries[i] = res.Query
+		}
+
+		query := strings.Join(queries, "\n")
+		err := alerta.NewClient(
```

Comment:
Ideally, we don't want to create `alerta.NewClient` every time in the method. We should create the client in the "main" function and pass it down with Dependency injection. (Or else use a factory and DI the factory, but I don't think a factory is needed)

A few issues:
* We will create a new HTTP client pool every time, and not reuse TCP connections, thus creating tcp connections every time.
* We can need to spin up a test http server to test it. If you used a simple interface, testing would be easier.